### PR TITLE
document request handling flow

### LINKED ISSUE #70: document request handling flow
/assign

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #70

### CODE REVIEW

**Comment by BenTheElder** [comment]
rendered: https://github.com/kubernetes-sigs/oci-proxy/blob/e45b27d56417f659c1fd401c23c498a861fae98c/cmd/archeio/docs/request-handling.md

**Comment by BenTheElder** [comment]
cc @dims

File: cmd/archeio/docs/request-handling.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Request Handling
+
+Requests to archeio follows the following flow:
+
+1. If it's a request for `/`: redirect to our wiki page about the project
+2. If it's not a request for `/` and does not start with `/v2/`: 404 error
+3. For registry API requests, all of which start with `/v2/`:
+  - If it's not a blob request: redirect to GCR
+  - If it's not a known AWS IP: redirect to GCR
+  -  If it's a known AWS IP AND HEAD reuqest for the layer succeeeds in S3: redirect to S3
```
Comment (dims) [inline cmd/archeio/docs/request-handling.md:10]: typo `reuqest`

File: cmd/archeio/docs/request-handling.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Request Handling
+
+Requests to archeio follows the following flow:
+
+1. If it's a request for `/`: redirect to our wiki page about the project
+2. If it's not a request for `/` and does not start with `/v2/`: 404 error
+3. For registry API requests, all of which start with `/v2/`:
+  - If it's not a blob request: redirect to GCR
+  - If it's not a known AWS IP: redirect to GCR
```
Comment (ameukam) [inline cmd/archeio/docs/request-handling.md:9]:
```suggestion
  - If it's not a known AWS IP: redirect to [k8s.gcr.io](https://k8s.gcr.io)
```

File: cmd/archeio/docs/request-handling.md (hunk 1 above)
Comment (ameukam) [inline cmd/archeio/docs/request-handling.md:10]:
```suggestion
  -  If it's a known AWS IP AND HEAD request for the layer succeeds in S3: redirect to the S3 bucket
```

File: cmd/archeio/docs/request-handling.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  -  If it's a known AWS IP AND HEAD reuqest for the layer succeeeds in S3: redirect to S3
+  -  If it's a known AWS IP AND HEAD fails: redirect to GCR
```
Comment (ameukam) [inline cmd/archeio/docs/request-handling.md:11]:
```suggestion
  -  If it's a known AWS IP AND HEAD fails: redirect to [k8s.gcr.io](https://k8s.gcr.io)
```

**Comment by dims** [comment]
one nit from me.! looks good in general. would be good to say k8s.gcr.io instead of GCR too.

**Comment by BenTheElder** [comment]
let's say the upstream registry throughout the flow and note separately for us that is k8s.gcr.io currently.

**Comment by BenTheElder** [comment]
rendered https://github.com/kubernetes-sigs/oci-proxy/blob/5fb8a372e29d53f00956b8d689996dbf35e6df6e/cmd/archeio/docs/request-handling.md

**Pull request merged**